src/server/util/API.js

```diff
@@ -53,6 +53,28 @@ async function checkRateLimit() {
     }
 }
 
+async function fetchRepositories(organization, page) {
+    const res = await get(APIHOST + `/orgs/${organization}/repos?per_page=100&page=${page}`)
+        if (res !== undefined) {
+            return res.data.map((element)=> {
+                return element["name"]
+        })} 
+        else {
+            return ''
+        }
+
+}
+
+async function getRepositories(organization) {
+    console.log('Called')
+    const results = []
+    for(var page =1; page<=3; page++)
```

Comment:
For orgs which have repos less than 100 will still be fetching for page2 and 3 over the network. We can avoid this and keep a count which acts as a flag whether we do want to make further requests or not. Suppose in a request for a specific page we get 0-99 repos, this indicates we do not need to make further requests.